casper/src/main/rholang/MakeMint.rho

```diff
@@ -0,0 +1,60 @@
+//scalapackage coop.rchain.rholang.mint
+
+//based on http://erights.org/elib/capability/ode/ode-capabilities.html#simple-money
+//requires NonNegativeNumber, MakeBrandPair
+contract @["MakeMint", "int2NN"](@x, return) = {
+  new nnCh in {
+    @"NonNegativeNumber"!(x, *nnCh) | for(@nn <- nnCh) {
+      return!(nn)
+    }
+  }
+} |
+contract @"MakeMint"(return) = {
+  new pairCh, thisMint, internalMakePurse in {
+    @"MakeBrandPair"!(*pairCh) | for(@p <- pairCh) {
+      match p {
+        [seal, unseal] => {
+          contract @[*thisMint, "makePurse"](@init, return) = {
+            new balanceCh in {
+              @["MakeMint", "int2NN"]!(init, *balanceCh) | for(@balance <- balanceCh) {
+                internalMakePurse!(balance, *return)
+              }
+            }
+          } |
+          contract internalMakePurse(@balance, return) = {
+            //balance must be a name that NonNegativeNumber contract is listening on
+            new thisPurse in {
+              contract @[*thisPurse, "decr"](@amount, success) = { @[balance, "sub"]!(amount, *success) } |
```

Comment:
I think you need `new thisPurse, decr` and then to use `decr` instead of `"decr"` because as it stands, you can just empty a purse on accident, and I don't think that's what you want.